bpf/lib/process.h

```diff
@@ -292,6 +292,23 @@ struct msg_execve_event {
 	__u32 binary;
 }; // All fields aligned so no 'packed' attribute.
 
+#define BINARY_PATH_MAX_LEN 256
+
+// This structure stores the binary path that was recorded on execve.
+// Technically PATH_MAX is 4096 but we limit the length we store since we have
+// limits on the length of the string to compare:
+// - Artificial limits for full string comparison.
+// - Technical limits for prefix and postfix, using LPM_TRIE that have a 256
+//   bytes size limit.
```

Comment:
Yes you are correct. This idea behind the first _artificial_ limit of 256 is that we align with LPM_TRIE limitation for what we can propose for the Prefix/Postfix. Also the length matters because the hash lookup will be orders of magnitude slower, the longer it is, because most of the time it will be filled with zeroes (so we would need some kind of optimization with variable size maps as we do with matchArgs and strings).

The idea is also that we need to store this path in execve_map, here we limit to 256 bytes so that for each process we have around 256 bytes more to store in maps. if we go to 4K, memory consumption will automatically max to 4K*number of processes.

Storing only the first 256 bytes is a bit annoying when we will need to do the Postfix, that's why in some cases (for process that have binary path len > 256) we will need to have 256 bytes for the beginning of the string and 256 bytes for the end of the string. For most processes this will be already overkill but this reduces the memory from 4096 to 512 for each process to do In/Prefix/Postfix.

I think it's already a high limit? In any case, if you need to match up to 256 you can use `Prefix`.

I checked on my host filesystem when I needed to choose between 128 and 256 (because hash lookup cost etc...) and I didn't find a lot of real-life files that had path > 128 chars. I guess if people suffer from this for their use case in the future we can always increase by switching the constants!

Reply:
Yes, I agree. I that we are fine with 256 bytes. My comment was mainly to check if I missed anything and just to keep in mind that if needed we can increase that further in the future. Thanks!